### PR TITLE
docs: added talos installation

### PR DESCRIPTION
**Pull request by balaharish7**

### CODE REVIEW

File: docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/openebs-on-kubernetes-platforms/talos.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        openebs.io/engine: mayastor
+    ```
+
+    - Run talosctl gen config with the above file.
+
+    ```
+    talosctl gen config my-cluster https://mycluster.local:6443 --config-patch @mayastor-patch.yaml
+    ```
+
+2. By patching an running node with config file:
```
Comment (avishnu) [inline docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/openebs-on-kubernetes-platforms/talos.md:46]: a running node

**Comment by avishnu** [comment]
@datacore-tilangovan how do we enable nvme-tcp module ?

**Comment by datacore-tilangovan** [comment]
> @datacore-tilangovan how do we enable nvme-tcp module ?

@avishnu, We dont have to enable it manually. they are enabled by default. It seems to work fine without adding nvme_tcp module. I have followed this doc by talos, and they are not loading nvme separately. But its not getting listed in loaded modules when we try "talosctl read /proc/modules"

https://www.talos.dev/v1.7/kubernetes-guides/configuration/storage/#mayastor

But there are users still reporting, loading nvme module manually is not working as expected. 
For example: https://discuss.kubernetes.io/t/talos-linux-cluster-missing-module/27519

I have tried the same but the modules are not listing in "talosctl read /proc/modules". Could be with "nvme-tcp" instead of "nvme_tcp" ? not sure. trying both.

**Comment by balaharish7** [comment]
@avishnu Any updates on this?

File: docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/openebs-on-kubernetes-platforms/talos.md [hunk 2]
```diff
@@ -10,45 +10,85 @@ description: This section explains about the Platform Support for Replicated PV
...
+
+## Talos Worker Node Changes
+
+### Huge Pages
+
+2MiB-sized Huge Pages must be supported and enabled on the Replicated PV Mayastor storage nodes. A minimum number of 1024 such pages (i.e. 2GiB total) must be available exclusively to the  Replicated PV Mayastor pod on each node.
+
+### Labels
+
+The OpenEBS storage type "Replicated PV Mayastor" must be labeled to all worker nodes that will have IO engine pods running on them. This label will be used as a node selector by the IO engine Daemonset that is deployed as a part of the Replicated PV Mayastor data plane components installation.
```
Comment (avishnu) [inline docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/openebs-on-kubernetes-platforms/talos.md:44]: The label is not "Replicated PV Mayastor". It is infact, 
_openebs.io/engine: "mayastor"_

**Comment by datacore-tilangovan** [comment]
> @datacore-tilangovan how do we enable nvme-tcp module ?

@avishnu @tiagolobocastro  I have discussed this with Talos over slack, they said these modules are inbuilt with kernel and statically compiled.

**Review by avishnu — APPROVED** [review]

**Pull request merged**